pkg/scheduler/framework/plugins/nodename/node_name.go

```diff
@@ -42,8 +45,16 @@ func (pl *NodeName) Name() string {
 
 // Filter invoked at the filter extension point.
 func (pl *NodeName) Filter(ctx context.Context, _ *framework.CycleState, pod *v1.Pod, nodeInfo *nodeinfo.NodeInfo) *framework.Status {
-	_, reasons, err := predicates.PodFitsHost(pod, nil, nodeInfo)
-	return migration.PredicateResultToFrameworkStatus(reasons, err)
+	node := nodeInfo.Node()
+	if node == nil {
+		return framework.NewStatus(framework.Error, "node not found")
+	}
+
+	if len(pod.Spec.NodeName) == 0 || pod.Spec.NodeName == node.Name {
+		return nil
+	}
```

Comment:
I feel like we can eliminate the duplication (with "podFitsHost()" in predicates.go) by move the core checking logic somewhere?

Reply:
> Thanks @ahg-g . Comments below.
> 
> The major concern is that we are duplicating the logic which serves for external components and scheduler plugins, IMO we should keep one copy.

Yes, but the duplication is for a simple piece of code. In any case, what I want to In a follow up PR is call the plugins directly from here.

Reply:
I am moving the code the plugins, will update the PR.